lua/lualine.lua

```diff
@@ -34,7 +34,19 @@ local function load_components()
     for _, section in pairs(sections) do
       for index, component in pairs(section) do
         if type(component) == 'string' then
-          section[index] = require('lualine.components.' .. component)
+          local ok,loaded_component = pcall(require, 'lualine.components.' .. component)
+          if not ok then
+            loaded_component = function()
+              if vim.fn.exists('*'..component) ~= 0 then
+                return vim.fn[component]() 
+              elseif vim.g[component] then
```

Comment:
This currently only supports global variables.

Reply:
We could add b: w: t: . Does v: make any sence ?

Reply:
Well... some variables are both related to window and global. There should probably be a correct order to checking them.

As of `vim.v` I have no idea how it works, always displays empty table no matter what I do.

Reply:
I think we should stick with globals only .

Reply:
>There should probably be a correct order to checking them.

If we are talking about prder then it clearly should be b > w > t .
But what to do when they are not available.

>As of vim.v I have no idea how it works, always displays empty table no matter what I do.

try `:help v:var`
vim.v / vim.g these tables are lazy_loading in lua . they don't store a key before it's used first time